src/psyclone/dynamo0p3.py

```diff
@@ -65,10 +65,10 @@
 #
 # ---------- Function spaces (FS) ------------------------------------------- #
 # Discontinuous FS
-DISCONTINUOUS_FUNCTION_SPACES = ["w3", "wtheta", "w2v"]
+DISCONTINUOUS_FUNCTION_SPACES = ["w3", "wtheta", "w2v", "w2broken"]
 # Continuous FS
-# Space any_w2 can be w2, w2h or w2v
-CONTINUOUS_FUNCTION_SPACES = ["w0", "w1", "w2", "w2h", "any_w2"]
+# Space any_w2 can be w2, w2h, w2v, w2broken or w2trace
+CONTINUOUS_FUNCTION_SPACES = ["w0", "w1", "w2", "w2h", "w2trace", "any_w2"]
```

Comment:
Yes, you are right. I think we placed it into the continuous list for the sake of simplicity as it must be treated as continuous due to at least one of `W2*` varieties being continuous. I clarified that by updating the comment.

In LFRic trunk `ANY_W2`,  `ANY_SPACE` and `ANY_DISCONTINUOUS_SPACE` are in [`argument_mod.F90`](https://code.metoffice.gov.uk/trac/lfric/browser/LFRic/trunk/infrastructure/source/field/argument_mod.F90) whereas all "real" function spaces now live in [fs_continuity_mod.F90](https://code.metoffice.gov.uk/trac/lfric/browser/LFRic/trunk/infrastructure/source/function_space/fs_continuity_mod.F90).

Based on the above, I think that `ANY_W2` should really be in some kind of "any_*_space" list (in fact, I think we should actually call it `ANY_W2_SPACE`). I also think it would be good to update the PSyclone version of LFRic infrastructure to reflect the "real FS" placement in a separate module. I am happy to create relevant issue(s) if required.

Reply:
**Update:** `ANY_W2` actually stands for any *vector* `W2*` space, that is `W2`, `W2H`, `W2V` and `W2broken` but not `W2trace` (chat with Tom) so I clarified this in  `dynamo0p3.py` and  `dynamo0p3.rst` . I also corrected the dimension of `ANY_W2` diff basis to 1  in the relevant table in `dynamo0p3.rst` (it is fine in the `dynamo0p3.py`).

Reply:
I agree with the name change suggestion and given your latest comment it should probably be ANY_W2_VECTOR_SPACE. I also agree that it should be in a separate list. Whether we need to extend to _1, _2 etc is up to the users. Tom seemed happy with one name and a restriction that the use of ANY_W2 multiple times in a kernel implies they are all on the same (unknown W2) space. I've created issue #540 rather than asking you to :-).